packages/apps/plugins/plugin-thread/src/ThreadPlugin.tsx

```diff
@@ -221,6 +223,25 @@ export const ThreadPlugin = (): PluginDefinition<ThreadPluginProvides> => {
               state.focus = intent.data?.focus;
               return { data: true };
             }
+
+            case ThreadAction.DELETE: {
+              const { document: doc, thread, cursor } = intent.data ?? {};
+              if (!isDocument(doc) || !isThread(thread)) {
+                return;
+              }
+
+              if (intent.inverse && typeof cursor === 'string') {
```

Comment:
it feels a bit error prone to `else` here if EITHER of these are false. feels like instead cursor === string should be an invariant inside the inverse (undo) case?